util/ratelimits.go

```diff
@@ -58,25 +58,30 @@ func (c *PostRequestCounter) ApiResponseRatelimitCounter(resp *resty.Response) e
 	if err != nil {
 		return err
 	}
+
+	// TODO: remove when rate-limits are simplified
+	currTime := time.Now().Unix()
+	secondaryLimitRefreshTime := c.RefreshTime - int(reconciler.SecondaryLinodeTooManyPOSTRequestsErrorRetryDelay.Seconds())
+	if c.ReqRemaining >= reconciler.SecondaryPOSTRequestLimit && currTime <= int64(secondaryLimitRefreshTime) {
+		c.RefreshTime = secondaryLimitRefreshTime
+	}
 	return nil
 }
 
 // IsPOSTLimitReached checks whether POST limits have been reached.
 func (c *PostRequestCounter) IsPOSTLimitReached() bool {
 	// TODO: Once linode API adjusts rate-limits, remove secondary rate limit and simplify accordingly
 	currTime := time.Now().Unix()
+	return currTime <= int64(c.RefreshTime) && (c.ReqRemaining == 0 || c.ReqRemaining == reconciler.SecondaryPOSTRequestLimit)
```

Comment:
```suggestion
	return time.Now().Unix() <= int64(c.RefreshTime) && (c.ReqRemaining == 0 || c.ReqRemaining == reconciler.SecondaryPOSTRequestLimit)
```
Also would add a comment explaining why the `c.ReqRemaining == reconciler.SecondaryPOSTRequestLimit` check is needed

Reply:
Sure, let me update it. Basically if we have made 5 requests (5 remaining) or 10 requests (0 remaining), then we want to wait until rate-limit has reset for that window.

Reply:
Updated